web-client/pages/survey.tsx

```diff
@@ -71,60 +243,87 @@ export default function Survey() {
-                    marginTop={10}
                 >
-                    <LinkButton href="/" text={'Back to home page'} />
-
-                    <Text fontSize="4xl">Survey placeholder</Text>
-                    <Text>
-                        Question {questionIndex + 1} / {questions.length}
-                    </Text>
-                    <Box maxWidth={'600px'}>
-                        <Text fontSize={'2xl'} fontWeight={'bold'}>
+                    <Progress
+                        borderRadius={12}
+                        minWidth={300}
+                        value={questionIndex + 1}
+                        max={questions.length}
+                        size="md"
+                        colorScheme="blue"
+                    />
+                    <Box
+                        maxWidth={'600px'}
+                        display={'flex'}
+                        flexDirection={'column'}
+                        justifyContent={'center'}
+                    >
+                        <Text
+                            fontSize={'2xl'}
+                            textAlign={'center'}
+                            fontWeight={'bold'}
+                        >
                             {questions[questionIndex].question}
                         </Text>
                         {questions[questionIndex].description && (
-                            <Text fontSize={'md'}>
+                            <Text fontSize={'md'} textAlign={'center'}>
                                 {questions[questionIndex].description}
                             </Text>
                         )}
                     </Box>
-                    {questions[questionIndex].answerType === 'text' ? (
-                        <Textarea />
-                    ) : (
-                        <Code> Answer Placeholder</Code>
-                    )}
+                    <Box>{answerComponents[questionIndex]}</Box>
                 </Box>
                 <Box>
-                    <Box display={'flex'} gap={5}>
+                    <Box display={'flex'} marginTop={10} gap={5}>
                         <Button
                             disabled={questionIndex === 0}
                             onClick={previousQuestion}
                         >
                             Previous question
                         </Button>
-
                         {questionIndex === questions.length - 1 ? (
-                            <LinkButton
-                                onClick={() => dispatch(setConfigState(true))}
-                                colorTheme="red"
-                                href="/"
-                                text="Finish survey"
-                            />
+                            <Button
+                                onClick={() => {
+                                    postForm({
+                                        name: 'user',
+                                        sports: Number.parseInt(sports),
```

Comment:
should there be some handling for if somehow `sports` isn't `Number`-looking?  ie returning `-1` or something when `parseInt` returns `NaN`